### ISSUE TITLE
Disable "Solidity Language & Themes" only

### ISSUE THREAD

**Issue by 0xpolarzero** [Consensys/vscode-solidity-auditor#140]
I would like to disable the extension "Solidity Language & Themes (only)", that is being installed with Visual Developer. Unfortunately, I can't:

`Cannot disable 'Solidity Language & Themes (only)' extension alone. 'Solidity Visual Developer' extension depends on this. Do you want to disable all these extensions?`

I find it very annoying, as for instance it started removing highlighting from Natspecs tags (e.g. `@notice`), although they are well-defined and work perfectly when disabling the whole extension suite.

<img width="274" alt="image" src="https://github.com/Consensys/vscode-solidity-auditor/assets/99199454/88ca1048-be47-4a8f-960a-9fcda92e651c">

This is the only syntax highlighting that seems to be overriden by the extension. I obviously tried redefining the custom highlighting in user settings, but nothing seems to do the trick.

Any way to still disable the language & themes extension? Or a trick at least?

Thanks!

**Comment by tintinweb** [comment]
`ctrl + shift + p -> inspect editor tokens and scope` - hover over `@notice` and post a screenshot here. for ref, this is mine. It shows a couple of scopes and depending on your theme they will be highlighted differently (I'm using the theme provided with this extension)

<img width="729" alt="image" src="https://github.com/Consensys/vscode-solidity-auditor/assets/2865694/298454b8-bd2e-41bf-9ae6-dd216143e708">

**Comment by 0xpolarzero** [comment]
@tintinweb Thanks for answering.

Well I actually did that already, with the same scopes, but can't seem to get the same result anymore. See screenshot.

<img width="580" alt="image" src="https://github.com/Consensys/vscode-solidity-auditor/assets/99199454/464cc4f6-7f3e-4d5c-8c0c-2b56c4c458d9">

But regardless, I already re-defined these in my user settings file, just in case. And I eventually found out that the syntax highlighting works in some cases, for some files, and that I can tweak it.

For instance, in this library (`lib/solady/src/tokens/ERC20.sol`).

<img width="489" alt="image" src="https://github.com/Consensys/vscode-solidity-auditor/assets/99199454/dffb1c30-0283-4bed-97e7-bca2bd63749e">

Or there (`lib/forge-std/src/interfaces/IERC20.sol`).

<img width="581" alt="image" src="https://github.com/Consensys/vscode-solidity-auditor/assets/99199454/4eb41bcd-88f2-4b8b-a489-433339a7086b">

But none work in my root `src` file.

**Comment by tintinweb** [comment]
in your screenshot with the texmate profile it does not seem like the solidity language extension is installed, see missing [storage.type.dev.natspec](https://github.com/tintinweb/vscode-solidity-language/blob/master/src/syntaxes/solidity.tmLanguage.json#L109). I cannot really reproduce it here. can you check your theme's spec as to what `@dev` would be highlighted as and are there any other extensions installed that provide texmate labeling for natspec?

def. weird. also that it only highlights sometimes in your screenshot. what does the texmate profiler say if you hover over the highlighted and non highlighted one? must be different labels.

**Comment by 0xpolarzero** [comment]
@tintinweb Indeed, I use [the language extension from Nomic Foundation](https://marketplace.visualstudio.com/items?itemName=NomicFoundation.hardhat-solidity).

And you're right, that's definitely the issue:
<img width="549" alt="image" src="https://github.com/Consensys/vscode-solidity-auditor/assets/99199454/f37ca554-6a01-4a98-911b-1f35a11bafd4">
<img width="596" alt="image" src="https://github.com/Consensys/vscode-solidity-auditor/assets/99199454/d3c5438b-68ea-4dfe-8149-efa279590b99">

Although as soon as I disable "Solidity Visual Developer", then "Solidity Language & Themes (only)", it starts working again without any issue.

<img width="522" alt="image" src="https://github.com/Consensys/vscode-solidity-auditor/assets/99199454/16b3b379-8ccb-46c4-aa79-16eee9a5dbdb">
